spring-integration-jmx/src/main/java/org/springframework/integration/jmx/config/MBeanExporterHelper.java

```diff
@@ -97,9 +114,19 @@ public Object postProcessBeforeInitialization(Object bean, String beanName) thro
 			@SuppressWarnings("unchecked")
 			Set<String> excludedNames = (Set<String>) mbeDfa.getPropertyValue(EXCLUDED_BEANS_PROPERTY_NAME);
 			if (excludedNames != null) {
-				siBeanNames.addAll(excludedNames);
+				excludedNames = new HashSet<String>(excludedNames);
+			}
+			else {
+				excludedNames = new HashSet<String>();
+			}
+			if (!this.hasIntegrationExporter) {
+				excludedNames.addAll(this.si42ChannelBeanNames);
+			}
+			else {
+				excludedNames.addAll(this.siBeanNames);
 			}
-			mbeDfa.setPropertyValue(EXCLUDED_BEANS_PROPERTY_NAME, siBeanNames);
+			//TODO: SF 4.1.5 and above now has additive exclusions (SPR-12686)
```

Comment:
That's not currently reflected in build.gradle; what drives that?

All I am saying with this comment is it's not available in 4.1.4 which is the current version in b.g.